### PR TITLE
Adding useFocusable to useLink so TooltipTrigger works with Link

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes https://github.com/adobe/react-spectrum/issues/2275

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Go to the "tooltip-on-link" story in TooltipTrigger and confirm that the tooltip appears above the link on hover/focus

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-spectrum/menu/src/ActionMenu.tsx [hunk 1]
```diff
@@ -39,11 +39,8 @@ function ActionMenu<T extends object>(props: SpectrumActionMenuProps<T>, ref: Fo
       shouldFlip={props.shouldFlip}>
       <ActionButton
         ref={ref}
-        {...buttonProps}
-        UNSAFE_className={props.UNSAFE_className}
-        isDisabled={props.isDisabled}
-        isQuiet={props.isQuiet}
-        autoFocus={props.autoFocus}>
+        {...props}
```
Comment (snowystinger) [inline packages/@react-spectrum/menu/src/ActionMenu.tsx:42]: why is this in your PR?

File: packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx [hunk 2]
```diff
@@ -197,6 +198,10 @@ storiesOf('TooltipTrigger', module)
         </Flex>
       </Flex>
     )
+  )
+  .add(
```
Comment (snowystinger) [inline packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx:202]: even better, lets add a test for this
I am doing that for ActionMenu right now as well

File: packages/@react-spectrum/menu/src/ActionMenu.tsx (hunk 1 above)
Comment (LFDanLu) [reply packages/@react-spectrum/menu/src/ActionMenu.tsx:42]: weird, no idea... Lemme remerge

File: packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx (hunk 2 above)
Comment (LFDanLu) [reply packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx:202]: will do

File: packages/@react-aria/link/src/useLink.ts [hunk 3]
```diff
@@ -56,14 +57,15 @@ export function useLink(props: AriaLinkOptions, ref: RefObject<HTMLElement>): Li
       tabIndex: !isDisabled ? 0 : undefined
     };
   }
-
+  let {focusableProps} = useFocusable(props, ref);
```
Comment (snowystinger) [inline packages/@react-aria/link/src/useLink.ts:60]: I'm almost surprised that FocusRing isn't the one to handle this?

Comment (LFDanLu) [reply packages/@react-aria/link/src/useLink.ts:60]: Well, there is the case where the user is making their own Link component and may not use `FocusRing` so we'd really wanna handle the `useFocusable` call in our aria hooks where possible IMO

**Review by reidbarber — APPROVED** [review]
LGTM. Tested on Chrome, Firefox, Safari

**Review by snowystinger — APPROVED** [review]
lgtm

**Pull request merged**